File: csharp/LogicAndTrick.WikiCodeParser/Models/WikiRevisionCredit.cs
namespace LogicAndTrick.WikiCodeParser.Models
{
    public class WikiRevisionCredit
    {
        public const char TypeCredit = 'c';
        public const char TypeArchive = 'a';
        public const char TypeFull = 'f';

        public int ID { get; set; }
        public char Type { get; set; }
        public int RevisionID { get; set; }
        public string Description { get; set; }
        public int? UserID { get; set; }
        public string Name { get; set; }
        public string Url { get; set; }
        public string WaybackUrl { get; set; }
    }
}

File: php/src/Models/WikiRevisionCredit.php
<?php

namespace LogicAndTrick\WikiCodeParser\Models;

class WikiRevisionCredit
{
    public const TypeCredit = 'c';
    public const TypeArchive = 'a';
    public const TypeFull = 'f';

    public ?int $ID = null;
    public string $Type;
    public ?int $RevisionID = null;
    public ?string $Description = null;
    public ?int $UserID = null;
    public ?string $Name = null;
    public ?string $Url = null;
    public ?string $WaybackUrl = null;
}